### PR TITLE
fix(list-services): Use state for finding status of list services

### PR DESCRIPTION
**Pull request by hubertdeng123**
For `list-services`, we are incorrectly marking local services as running when services are brought up with remote dependencies. 

Say for example snuba is brought up that depends on shared-redis and shared-kafka whose definitions are pulled in remotely. `devservices list-services` would show local copies of shared-redis and shared-kafka as running. This fixes that

### CODE REVIEW

File: devservices/utils/docker_compose.py [hunk 1]
```diff
@@ -29,21 +29,6 @@
 from devservices.utils.services import Service
 
 
-def get_active_docker_compose_projects() -> list[str]:
```
Comment (hubertdeng123) [inline devservices/utils/docker_compose.py:32]: 👋

File: tests/commands/test_list_services.py [hunk 2]
```diff
@@ -7,20 +7,19 @@
...
-    mock_run: mock.Mock, tmp_path: Path, capsys: pytest.CaptureFixture[str]
+    tmp_path: Path, capsys: pytest.CaptureFixture[str]
 ) -> None:
     with mock.patch(
         "devservices.commands.list_services.get_coderoot",
         return_value=str(tmp_path / "code"),
-    ):
+    ), mock.patch("devservices.utils.state.STATE_DB_FILE", str(tmp_path / "state")):
+        state = State()
+        state.add_started_service("example-service", "default")
```
Comment (IanWoodard) [inline tests/commands/test_list_services.py:22]: Feel free to punt this for now, but ideally we should be calling start (not necessarily actually starting the service, but doing whatever state and other misc things that are required)

File: tests/commands/test_list_services.py [hunk 3]
```diff
@@ -52,17 +51,13 @@ def test_list_running_services(
...
-    mock_run: mock.Mock, tmp_path: Path, capsys: pytest.CaptureFixture[str]
-) -> None:
+def test_list_all_services(tmp_path: Path, capsys: pytest.CaptureFixture[str]) -> None:
     with mock.patch(
         "devservices.commands.list_services.get_coderoot",
         return_value=str(tmp_path / "code"),
-    ):
+    ), mock.patch("devservices.utils.state.STATE_DB_FILE", str(tmp_path / "state")):
+        state = State()
+        state.add_started_service("example-service", "default")
```
Comment (IanWoodard) [inline tests/commands/test_list_services.py:60]: Same comment as above

**Review by IanWoodard — APPROVED** [review]

File: tests/commands/test_list_services.py (hunk 2 above)
Comment (hubertdeng123) [reply tests/commands/test_list_services.py:22]: seems out of scope for now but can consider in the future

**Pull request merged**